src/WebJobs.Script.WebHost/App_Start/AutofacBootstrap.cs

```diff
@@ -22,12 +23,18 @@ internal static void Initialize(ContainerBuilder builder, WebHostSettings settin
                 FileLoggingEnabled = true
             };
 
-            // If there is an explicit machine key, it makes a good default host id. It can still be
-            // overridden in host.json
-            var section = (MachineKeySection)ConfigurationManager.GetSection("system.web/machineKey");
-            if (section.Decryption != "Auto" && section.ValidationKey.Length >= 32)
+            // If running on Azure Web App, derive the host ID from the site name
+            string hostId = Environment.GetEnvironmentVariable("WEBSITE_SITE_NAME");
+            if (!String.IsNullOrEmpty(hostId))
             {
-                scriptHostConfig.HostConfig.HostId = section.ValidationKey.Substring(0, 32).ToLowerInvariant();
+                // Truncate to the max host name length if needed
+                const int MaximumHostIdLength = 32;
+                if (hostId.Length > MaximumHostIdLength)
+                {
+                    hostId = hostId.Substring(0, MaximumHostIdLength);
```

Comment:
there is a problem Phil found with this approach, if the site name after truncation ended up with a '-', although this is a valid character in queue name, it can not be the last character! there was another suggestion over Email